### ISSUE TITLE
[BUG] defining colors is inconsistent

### ISSUE THREAD

**Issue by tcmetzger** [bokeh/bokeh#10902]
### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
Bokeh 2.3.0dev13 (Python 3.8.6, Win10)

### Description of expected behavior and the observed behavior
Expected: Defining and using colors should be consistent across all different ways to define colors (as much as possible)
Observed: Some ways of defining colors work different than others

### Complete, minimal, self-contained example code that reproduces the issue

**Example 1: defining single colors with CSS color strings**

A single `rgb()` color string works:
```
from bokeh.plotting import figure, show

x = [1,2,3]
y = [1,2,3]
col = "rgb(200 0 0)"

p = figure(title = "Using single rgb() color value works")
p.circle(x, y, radius=.25, color=col)
show(p)
```
However, a single `hsl()` color string does not work:
```
from bokeh.plotting import figure, show

x = [1,2,3]
y = [1,2,3]
col = "hsl(60deg 100% 50% / 1.0)"

p = figure(title = "Using single hsl() color value fails")
p.circle(x, y, radius=.25, color=col)
show(p)
```
hsl() color strings as items in a list / array do work, therefore I guess that adding a hsl regex to `__init__()` in Bokeh.core.properties.Color might resolve this problem.

**Example 2: defining single colors with a 32-bit unsigned integer**

Similarly, using a single 32-bit unsigned integer does not work:
```
from bokeh.plotting import figure, show

x = [1,2,3]
y = [1,2,3]
col = 0xff0000ff

p = figure(title = "Using single 32-bit int color fails")
p.circle(x, y, radius=.25, color=col)
show(p)
```
Also, 32-bit unsigned integers as items in a list do work, so the rexpective data type might have to be added to `__init__()` in Bokeh.core.properties.Color 

**Example 3: Inconsistent use of commas in CSS color strings**

It is possible to express CSS color strings either with or without commas (e.g. `"rgb(0, 0, 255)"` should be the same as `"rgb(0 0 255)"`)

This works for ``rgb()``:
```
from bokeh.plotting import figure, show

x = [1,2,3]
y1 = [1,2,3]
y2 = [3,1,2]
color_with_comma = "rgb(200, 0, 0)"
color_without_comma = "rgb(0 200 0)"

p = figure(title = "rgb() works with or without commas")
p.circle(x, y1, size=25, color=color_with_comma)
p.triangle(x,y2, size=25, color=color_without_comma)
show(p)
```
However, for ``rgba()`` only strings _with_ commas work:

```
from bokeh.plotting import figure, show

x = [1,2,3]
y1 = [1,2,3]
y2 = [3,1,2]
color_with_comma = "rgba(255, 0, 0, 0.5)"
color_without_comma = "rgba(0 255 0 0.5)"

p = figure(title = "rgba() only works with commas")
p.circle(x, y1, size=25, color=color_with_comma)
p.triangle(x,y2, size=25, color=color_without_comma)
show(p)
```
This could potentially also be fixed by updating the regexes Bokeh.core.properties.Color.

**Comment by mattpap** [comment]
> `color_without_comma = "rgba(0 255 0 0.5)"`

This is incorrect. CSS4 comma-less syntax requires `/` before alpha value (see https://www.w3.org/TR/css-color-4/#rgb-functions).

**Comment by mattpap** [comment]
In general, validation in bokeh needs to be updated, to match bokehjs' capability. However, that's nontrivial, because when assigning `color="some_str"`, it can mean either `color=field("some_str")` or `color=value("some_str")`, depending on the contents of `"some_str"` (see `ColorSpec.to_serializable()` and others). What's worse, bokehjs will happily accept any string as a color (there's a lot more than `rgb[a]()` in CSS4 spec, but only `rgb[a]()` is handled independently of browser support), which if supported, would mean that we wouldn't be able pass color fields at all.
